### ISSUE TITLE
Invalid record when running llvm-dis

### ISSUE THREAD

**Issue by bohde** [wmoses/Parallel-IR#56]
I was trying to test out the Tapir using the command line tools, and whenever I use `llvm-dis`, I get the following error:

```
llvm-dis: error: Invalid record (Producer: 'LLVM5.0.0git-ec3ad2b' Reader: 'LLVM 5.0.0git-ec3ad2b')
```

I'm using version ec3ad2b8d3810dde9c0aaccf3f3f971144d90bc2, and using the following command to replicate it:

```
llvm-as nloop.ll -o=- | llvm-dis
```

The nloop.ll file is below: 

```
declare token @llvm.syncregion.start() #3

declare void @print(i32)

define void @test (i32 %n) {
entry:
    %i = alloca i32
    store i32 0, i32* %i  ; store it back
    br label %loop.1.entry

loop.1.entry:
    %tmp = load i32, i32* %i
    %cmp = icmp ne i32 %tmp, %n
    br i1 %cmp, label %loop.1.body, label %loop.1.exit

loop.1.body:
    %syncreg = call token @llvm.syncregion.start()
    detach within %syncreg, label %loop.1.body.1, label %loop.1.body.2

loop.1.body.1:
    %tmp2 = load i32, i32* %i
    call void @print(i32 %tmp2)
    reattach within %syncreg, label %loop.1.body.2

loop.1.body.2:
    %tmp3 = load i32, i32* %i
    %tmp4 = add i32 %tmp3, 1   ; increment it
    store i32 %tmp4, i32* %i  ; store it back
    br label %loop.1.final

loop.1.final:
    sync within %syncreg, label %loop.1.entry

loop.1.exit:
    ret void
}
```

**Comment by neboat** [comment]
Thank you for the report.  I think I've identified the source of the bug, and I'm testing a fix now on the CI.

**Comment by wsmoses** [comment]
Resolved by #57
